src/webpage-digester.ts

```diff
@@ -86,10 +88,14 @@ export class WebPageDigester {
 
         // create new mainFilterInstance from this.searchTerms if not present
         if (!this.mainFilterInstance) {
-            this.mainFilterInstance = new this.mainFilter(this.searchTerms);
+            // construct a set from the map
+            let set : Set<string> = new Set();
+            for (let [term /*, id*/] of this.termToIDMap.entries())  set.add(term);
```

Comment:
_Wiihuu Wiihuu_ DUBLICATE CODE DETECTED! _Wiihuu Wiihuu_
Could you please make a little helper function for the conversion. Also please always use curly brackets for all `if` statements. Sorry, my OCD kicked in 😎